src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -1,8 +1,80 @@
 package com.zipcodewilmington.looplabs;
 
+import java.util.Arrays;
+
 /**
  * Created by leon on 1/29/18.
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
+
+    //constructor
+    public IntegerDuplicateDeleter (Integer[] intArray) {
+        super(intArray);
+    }
+
+
+    private static int countDuplicates (Integer[] array, Integer value) {
+        int dupeCount = 0;
+        for(int i = 0; i < array.length; i++) {
+            if(value.equals(array[i])) {
+                dupeCount++;
+            }
+        }
+        return dupeCount;
+    }
+
+
+
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+
+        Integer[] notYetDel = Arrays.copyOf(this.array, this.array.length);
+        String keepInteger = "";
+
+        for(Integer keepIt: notYetDel) {
+            int x = countDuplicates(notYetDel, keepIt);
+            if (x < maxNumberOfDuplications) {
+                keepInteger += keepIt + "/";
+            }
+
+        }
+        if(keepInteger.equals("")) {
+            keepInteger = "/";
+        }
+
+        String[] dupsRemoved = keepInteger.split("/");
+        Integer [] arrayReturned = new Integer [dupsRemoved.length];
+
+        for(int i = 0; i<arrayReturned.length; i++) {
+            arrayReturned[i] = Integer.parseInt(dupsRemoved[i]);
+        }
+
+        return arrayReturned;
```

Comment:
consider naming this `arrayWithDuplicatesRemoved` or something more descriptive than just `arrayReturned`. By being in the return statement anyone should be aware that this is the array returned. 

Passed that, this is a good answer. Converting from string to integer is a dangerous operation though so be careful about that.